### PR TITLE
Add CRC options when configuring Zoom integration

### PR DESCRIPTION
**Pull request by chemamartinez**
## What does this PR do?

This is an improvement of #6288 adding the following:
- Instead of hardcoding the CRC provider option for the Zoom integration, now it is exposed to users the option to enable it when configuring the integration.
- The reason of this change, is because a new option has been added called `crc.secret` at https://github.com/elastic/beats/pull/35604. This forces the user to add this secret token manually when setting up the integration.
- A new parameter `secret_header` has been added since now, Zoom allows verifying events with a custom header instead of using `Authorization` as a hardcoded one.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Depends on https://github.com/elastic/beats/pull/35604

## Screenshots

Screenshot 2023-05-31 at 19 19 50

image

### CODE REVIEW

File: packages/zoom/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.9.1"
```
Comment (efd6) [inline packages/zoom/changelog.yml:2]:
```suggestion
- version: "1.10.0"
```
since this is an enhancement.

File: packages/zoom/manifest.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 name: zoom
 title: Zoom
-version: "1.9.0"
+version: "1.9.1"
```
Comment (efd6) [inline packages/zoom/manifest.yml:3]:
```suggestion
version: "1.10.0"
```

File: packages/zoom/data_stream/webhook/manifest.yml [hunk 3]
```diff
@@ -29,10 +29,33 @@ streams:
...
+        title: Enable CRC validation
+        description: CRC validation is used by certain providers to verify the authenticity of a webhook endpoint.
+        multi: false
+        required: false
+        show_user: true
+        default: false
+      - name: crc_secret
+        type: text
+        title: Zoom Secret Token
+        description: Secret token provided by Zoom when configured the webhook. It is used for the CRC validation of the webhook endpoint.
```
Comment (efd6) [inline packages/zoom/data_stream/webhook/manifest.yml:43]:
```suggestion
        description: Secret token provided by Zoom when the webhook was configured. It is used for the CRC validation of the webhook endpoint.
```

File: packages/zoom/data_stream/webhook/manifest.yml [hunk 4]
```diff
@@ -29,10 +29,33 @@ streams:
...
+        multi: false
+        required: false
+        show_user: true
+      - name: secret_header
+        type: text
+        title: Zoom Custom Header
+        description: Custom header used to validate the authenticity of incoming Zoom POST requests. It should be created when configuring the webhook on Zoom. See [this documentation](https://developers.zoom.us/docs/api/rest/webhook-reference/#custom-header) for more information about this process.
```
Comment (efd6) [inline packages/zoom/data_stream/webhook/manifest.yml:50]: > "It should be created"

Is this "It should be created" by the user "when configuring the webhook", or "It should have been created" by Zoom "when configuring the webhook"?

File: packages/zoom/data_stream/webhook/manifest.yml [hunk 5]
```diff
@@ -29,10 +29,33 @@ streams:
...
+        multi: false
+        required: true
+        show_user: true
+        default: Authorization
       - name: secret_value
         type: text
-        title: Zoom verification token
-        description: Verification token used to validate the authenticity of incoming Zoom POST requests. Zoom sends this as the Authorization header.
+        title: Zoom Custom Header value
+        description: Custom header value used to validate the authenticity of incoming Zoom POST requests. It should be created when configuring the webhook on Zoom. See [this documentation](https://developers.zoom.us/docs/api/rest/webhook-reference/#custom-header) for more information about this process.
```
Comment (efd6) [inline packages/zoom/data_stream/webhook/manifest.yml:58]: Same query.

File: packages/zoom/data_stream/webhook/agent/stream/http_endpoint.yml.hbs [hunk 6]
```diff
@@ -2,16 +2,18 @@ listen_address: {{listen_address}}
...
-# Only set the secret options if a secret.value is set
-{{#if secret_value}}
-secret.header: Authorization
-secret.value: "{{secret_value}}"
+# Fill CRC settings if enabled
+{{#if crc_enabled}}
+crc.provider: zoom
+crc.secret: "{{crc_secret}}"
 {{/if}}
+# Secret header and value are mandatory for new integrations
```
Comment (efd6) [inline packages/zoom/data_stream/webhook/agent/stream/http_endpoint.yml.hbs:14]:
```suggestion
# Secret header and value are mandatory for new Zoom integrations
```
?

Comment (chemamartinez) [reply packages/zoom/data_stream/webhook/agent/stream/http_endpoint.yml.hbs:14]: Sorry, I should have been more clear with this. I meant to say that these parameters are now required when setting up Zoom integration.

File: packages/zoom/data_stream/webhook/manifest.yml (hunk 4 above)
Comment (chemamartinez) [reply packages/zoom/data_stream/webhook/manifest.yml:50]: By the user, I'll clarify that.

File: packages/zoom/data_stream/webhook/agent/stream/http_endpoint.yml.hbs [hunk 7]
```diff
@@ -2,16 +2,18 @@ listen_address: {{listen_address}}
...
-# Only set the secret options if a secret.value is set
-{{#if secret_value}}
-secret.header: Authorization
-secret.value: "{{secret_value}}"
+# Fill CRC settings if enabled
+{{#if crc_enabled}}
+crc.provider: zoom
+crc.secret: "{{crc_secret}}"
 {{/if}}
+# Secret header and value are required when configuring a new Zoom integration
```
Comment (P1llus) [inline packages/zoom/data_stream/webhook/agent/stream/http_endpoint.yml.hbs:14]: Secret header and value is optional settings, please add a if condition around them as well.

**Review by P1llus — APPROVED** [review]
One nit, il approve it and feel free to merge after :)

**Pull request merged**